src/simplewallet/simplewallet.cpp

```diff
@@ -8354,3 +8581,1025 @@ int main(int argc, char* argv[])
+  success_msg_writer() << tr("Content: ") << (display_content ? m.content : tr("(binary data)"));
+
+  if (m.type == mms::message_type::note)
+  {
+    // Showing a note and read its text is "processing" it: Set the state accordingly
+    // which will also delete it from Bitmessage as a side effect
+    // (Without this little "twist" it would never change the state, and never get deleted)
+    ms.set_message_processed_or_sent(m.id);
+  }
+}
+
+void simple_wallet::ask_send_all_ready_messages()
+{
+  mms::message_store& ms = m_wallet->get_message_store();
+  std::vector<mms::message> ready_messages;
+  const std::vector<mms::message> &messages = ms.get_all_messages();
+  for (size_t i = 0; i < messages.size(); ++i)
+  {
+    const mms::message &m = messages[i];
+    if (m.state == mms::message_state::ready_to_send)
+    {
+      ready_messages.push_back(m);
+    }
+  }
+  if (ready_messages.size() != 0)
+  {
+    list_mms_messages(ready_messages);
+    bool send = ms.get_auto_send();
+    if (!send)
+    {
+      send = user_confirms(tr("Send these messages now?"));
+    }
+    if (send)
+    {
+      mms::multisig_wallet_state state = get_multisig_wallet_state();
+      for (size_t i = 0; i < ready_messages.size(); ++i)
+      {
+        ms.send_message(state, ready_messages[i].id);
+        ms.set_message_processed_or_sent(ready_messages[i].id);
+      }
+      success_msg_writer() << tr("Sent.");
+    }
+  }
+}
+
+bool simple_wallet::get_message_from_arg(const std::string &arg, mms::message &m)
+{
+  mms::message_store& ms = m_wallet->get_message_store();
+  bool valid_id = false;
+  uint32_t id;
+  try
+  {
+    id = (uint32_t)boost::lexical_cast<uint32_t>(arg);
+    valid_id = ms.get_message_by_id(id, m);
+  }
+  catch (const boost::bad_lexical_cast &)
+  {
+  }
+  if (!valid_id)
+  {
+    fail_msg_writer() << tr("Invalid message id");
+  }
+  return valid_id;
+}
+
+void simple_wallet::mms_init(const std::vector<std::string> &args)
+{
+  if (args.size() != 3)
+  {
+    fail_msg_writer() << tr("usage: mms init <threshold>/<coalition_size> <own_label> <own_transport_address>");
+    return;
+  }
+  mms::message_store& ms = m_wallet->get_message_store();
+  if (ms.get_active())
+  {
+    if (!user_confirms(tr("The MMS is already initialized. Re-initialize by deleting all member info and messages?")))
+    {
+      return;
+    }
+  }
+  uint32_t threshold;
+  uint32_t coalition_size;
+  const std::string &mn = args[0];
+  std::vector<std::string> numbers;
+  boost::split(numbers, mn, boost::is_any_of("/"));
+  bool mn_ok = (numbers.size() == 2)
+               && get_number_from_arg(numbers[0], threshold, 2, 100)
+               && get_number_from_arg(numbers[1], coalition_size, 2, 100);
+  if (mn_ok)
+  {
+    mn_ok = (threshold <= coalition_size);
+    // Fully general cases like 3/5 now supported
+  }
+  if (!mn_ok)
+  {
+    fail_msg_writer() << tr("Error in threshold and/or coalition size");
+    return;
+  }
```

Comment:
Sorry, I don't get your proposal. Can you please elaborate?

Reply:
I had actually misread, but I mean instead of:
<pre>
  bool mn_ok = (numbers.size() == 2)
               && get_number_from_arg(numbers[0], threshold, 2, 100) 
               && get_number_from_arg(numbers[1], coalition_size, 2, 100);
  if (mn_ok)
  { 
    mn_ok = (threshold <= coalition_size);
    // Fully general cases like 3/5 now supported
  }
  if (!mn_ok)
  { 
    fail_msg_writer() << tr("Error in threshold and/or coalition size");
    return;
  }

<pre>,

<pre>
  bool mn_ok = (numbers.size() == 2)
               && get_number_from_arg(numbers[1], coalition_size, 2, 100) 
               && get_number_from_arg(numbers[0], threshold, 2, coalition_size);
  if (!mn_ok)
  { 
    fail_msg_writer() << tr("Error in threshold and/or coalition size");
    return;
  }
</pre>

seems simpler.

Reply:
Clever. I copied your version into the code.